src/main/java/com/mraof/minestuck/item/KeyItem.java

```diff
@@ -0,0 +1,75 @@
+package com.mraof.minestuck.item;
+
+import com.mraof.minestuck.Minestuck;
+import com.mraof.minestuck.block.*;
+import net.minecraft.client.util.ITooltipFlag;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemGroup;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.util.NonNullList;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraft.world.World;
+import net.minecraft.world.dimension.Dimension;
+import net.minecraftforge.common.util.Constants;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+public class KeyItem extends Item
+{
+	public EnumKeyType keyID;
+	//public Dimension dimension = null;
```

Comment:
Im hoping to add dimension specificity so keys only work on the land they were generated for. May need help working on that

Reply:
Can say that it should be stored in the item stack nbt, so you should probably write functions for the dimension and key type to read from/write to nbt. In 1.15, dimensions are referred to with an `DimensionType`.